hazelcast/src/main/java/com/hazelcast/client/impl/protocol/util/MessageFlyweight.java

```diff
@@ -52,8 +52,12 @@ public MessageFlyweight() {
         offset = 0;
     }
 
-    public MessageFlyweight wrap(ClientProtocolBuffer buffer, int offset) {
-        this.buffer = buffer;
+    public MessageFlyweight wrap(byte[] buffer, int offset, boolean useUnsafe) {
+        if (useUnsafe) {
```

Comment:
This would be an excellent place to use a ternery operator so you go from 5 lines to 1.